### PR TITLE
fixes on initialize & download

### PR DESCRIPTION
**Pull request by alexcos20**
A couple of fixes for errors encountered during ocean.js development

### CODE REVIEW

File: ocean_provider/routes/consume.py [hunk 1]
```diff
@@ -167,25 +167,26 @@ def initialize():
...
-        download_url = get_download_url(url_object, app.config["PROVIDER_CONFIG_FILE"])
-        download_url = append_userdata(download_url, data)
-        valid, url_details = check_url_details(download_url)
-
-        if not valid:
-            logger.error(
-                f"Error: Asset URL not found or not available. \n"
-                f"Payload was: {data}",
-                exc_info=1,
+        file_index = int(data.get("fileIndex", "-1"))
```
Comment (calina-c) [inline ocean_provider/routes/consume.py:170]: I would say just return an error message if file index is not ok, then un-indent the contents of the "if" statement

File: ocean_provider/routes/consume.py [hunk 2]
```diff
@@ -276,8 +277,10 @@ def download():
         )
 
         file_index = int(data.get("fileIndex"))
-
-        url_object = get_service_files_list(service, provider_wallet)[file_index]
+        files_list = get_service_files_list(service, provider_wallet)
+        if file_index > len(files_list):
+            return jsonify(error=f"No such fileIndex")
```
Comment (calina-c) [inline ocean_provider/routes/consume.py:282]: can you also add a 400 status code? And I would suggest a tighter condition on the if, e.g. if the file index is not found in the file list. Possibly it's `if file_index not in files_list` or `if file_index not in enum(files_list)`

File: ocean_provider/utils/url.py [hunk 3]
```diff
@@ -201,7 +201,7 @@ def _get_result_from_url(url, with_checksum=False):
 
 
 def append_userdata(url, data):
-    userdata = data.get("userdata")
+    userdata = data.get("userdata", None)
```
Comment (calina-c) [inline ocean_provider/utils/url.py:204]: yes, sorry for missing that :)

**Review by calina-c — CHANGES_REQUESTED** [review]

File: ocean_provider/routes/consume.py (hunk 1 above)
Comment (alexcos20) [reply ocean_provider/routes/consume.py:170]: this piece of code checks the validity of a file , as long as we have fileIndex defined.
If not, provider will return data for initialize, but without checking the file (maybe frontend already did that with fileinfo endpoint)

File: ocean_provider/routes/consume.py (hunk 2 above)
Comment (alexcos20) [reply ocean_provider/routes/consume.py:282]: agreed with 400.   but for tighter condition, it's enough.  files_list is an array, you cannot have files_list[0] and files_list[2] , but file_.list[1] is missing.

**Review by calina-c — APPROVED** [review]

**Pull request merged**